unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -55,13 +55,17 @@ public static UTF8String fromString(String str) {
   /**
    * Updates the UTF8String with String.
    */
-  public UTF8String set(final String str) {
-    try {
-      bytes = str.getBytes("utf-8");
-    } catch (UnsupportedEncodingException e) {
-      // Turn the exception into unchecked so we can find out about it at runtime, but
-      // don't need to add lots of boilerplate code everywhere.
-      PlatformDependent.throwException(e);
+  protected UTF8String set(final String str) {
+    if (str == null) {
+      bytes = new byte[0];
```

Comment:
It's safe to remove this check, because `set` is protected now, we can make sure that `str` is not null.